### ISSUE TITLE
No spacing between tags

### ISSUE THREAD

**Issue by yves-chevallier** [squidfunk/mkdocs-material#7548]
### Context

When using both `tags` and `search` plugins, the tags in the search bars are sticking to each others without proper spacing. I guess it can be fixed with the following:

```css
.md-typeset .md-tag {
    margin-right: 0.5em;
    margin-bottom: 0.5em;
}
```

### Bug description

In this example I have few tags. We notice both horizontally and vertically there is no margin:

![image](https://github.com/user-attachments/assets/46a87f03-5e6b-413c-b31b-865d12517270)

### Related links

https://squidfunk.github.io/mkdocs-material/setup/setting-up-tags/

### Reproduction

It happens with the latest insiders version

```toml
[tool.poetry]
name = "mkdocs-material-sandbox"
version = "0.1.0"
description = ""
readme = "README.md"

[tool.poetry.dependencies]
python = "^3.12"
mkdocs = "^1.6.1"
mkdocs-material = { git = "git@github.com:heig-tin-info/mkdocs-material-insiders.git" }

[build-system]
requires = ["poetry-core"]
build-backend = "poetry.core.masonry.api"
```

```yaml
site_name: My Docs
theme:
  name: material
plugins:
  - tags
  - search
```

```md
---
tags:
  - HTML5
  - JavaScript
  - CSS3
  - Web Development
  - Web Design
  - Web
  - Development
  - Design
  - Frontend
  - Front-end
  - Back to the Future
  - Marty McFly
---

# Welcome to MkDocs
```

### Steps to reproduce

Run mkdocs materials with few tags... 

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) using the [built-in info plugin](https://squidfunk.github.io/mkdocs-material/plugins/info/).

**Comment by squidfunk** [comment]
Thanks for reporting. Fixed in 06fe18af7. This is a regression of a tags refactoring we did some time ago. It also appeared in the community edition and is not exclusive to Insiders.

**Comment by yves-chevallier** [comment]
Your response time on this is nothing short of legendary. It's like you hot-patched it before the exception was even thrown—on a Sunday, no less. Super impressive!

**Comment by squidfunk** [comment]
Released as part of [9.5.37](https://squidfunk.github.io/mkdocs-material/changelog/#9.5.37).
